app/controllers/registers_controller.rb

```diff
@@ -23,6 +23,7 @@ def in_progress
   def show
     @register = Register.find_by_slug!(params[:id])
     @records = recover_records(@register.fields_array, params)
+    @is_empty_register = !@register.records.where(entry_type: 'user').any?
```

Comment:
Could keep the scope and replace this with `@is_empty_register = !@register.records.user_records.any?`

or maybe `@is_empty_register = @register.records.user_records.nil?`

Reply:
The problem with this is that the `user_records` scope was influenced by query parameters and filtering. The definition of whether a register is empty is different to whether `@register.records.user_records.nil?` is true. This could be true but it's not an empty register. It just happens that someone is filtering at the time. It seemed less confusing to separate the two variables - `@records` is influenced by filtering/searching but `@is_empty_register` is not. `@is_empty_register` can be false when the `@records` is empty.

Reply:
There was also a bug where the search bar disappears when you searched for something that resulted in no records. e.g. search for "bla" in the country register, the search box disappears.  This isn't possible to fix if `@is_empty_register` depends on `@records`.